frontend/app/utils/date-utils.ts

```diff
@@ -68,3 +102,20 @@ export function getAgeFromDateString(date: string) {
   invariant(isPast(parsedDate), `date must be in past [${date}]`);
   return differenceInYears(new Date(), parsedDate);
 }
+function parseDateValue(dateParts: string[], pattern: string): { day: number; month: number; year: number } {
+  // Have to add the '+' because that is how TS isNaN works apparently
+  if (dateParts.filter((element) => isNaN(+element)).length > 0) {
+    return { day: -1, month: -1, year: -1 };
+  }
+
+  // TODO: TO FILL OUT OTHER DATE FORMATS
+  if (pattern == 'yyyy/mm/dd' || pattern == 'yyyy-mm-dd' || pattern == 'yyyymmdd') {
+    return { day: Number.parseInt(dateParts[2]), month: Number.parseInt(dateParts[1]), year: Number.parseInt(dateParts[0]) };
+  } else if (pattern == 'dd/mm/yyyy' || pattern == 'dd-mm-yyyy' || pattern == 'ddmmyyyy') {
+    return { day: Number.parseInt(dateParts[0]), month: Number.parseInt(dateParts[1]), year: Number.parseInt(dateParts[2]) };
+  } else if (pattern == 'mm/dd/yyyy' || pattern == 'mm-dd-yyyy' || pattern == 'mmddyyyy') {
+    return { day: Number.parseInt(dateParts[1]), month: Number.parseInt(dateParts[0]), year: Number.parseInt(dateParts[2]) };
+  } else {
+    return { day: -1, month: -1, year: -1 };
```

Comment:
This caught me off-guard when I saw it.. I think you should consider not returning magic values if you can avoid it, but if you absolutely must, then add some JSDoc to the function explaining when these values will be returned and why.

Reply:
I prefer using the `-1` to catch with the date parsing validation but I did end up updating the JSDoc: 2ab2c0455279953194aad59443cee505be0bec8d. Let me know if it's sufficient.

Reply:
I think the jsdoc is *okay*. :)

It's acceptable, and I'll resolve my comment.. but I do have a small issue with:

> Checks if a string is a validad [sic] date. Else returns false.

then goes on to explain that it sometimes actually doesn't return false.

Here's the JSDoc I would write for this function:

``` typescript
/**
 * Parses a date string based on a given format pattern.
 *
 * This function takes a string array representing date components and a pattern string
 * indicating the order of those components. It attempts to parse the date based on the
 * provided pattern and returns an object with `day`, `month`, and `year` properties.
 *
 * @param {string[]} dateParts
 *   An array of strings representing the date components (e.g., `["2024", "05", "16"]`).
 *
 * @param {string} pattern
 *   A string representing the date format pattern (e.g., `"yyyy-mm-dd"`).
 * 
 *   Some example patterns include:
 *   - `"yyyy/mm/dd"`, `"yyyy-mm-dd"`, `"yyyymmdd"` (year, month, day)
 *   - `"dd/mm/yyyy"`, `"dd-mm-yyyy"`, `"ddmmyyyy"` (day, month, year)
 *   - `"mm/dd/yyyy"`, `"mm-dd-yyyy"`, `"mmddyyyy"` (month, day, year)
 *
 * @returns {object}
 *   An object containing the parsed date information:
 *     - `day` (number): The day of the month (1-31).
 *     - `month` (number): The month number (1-12).
 *     - `year` (number): The full year (e.g., 2024).
 * 
 *  ⚠️ If the parsing fails due to invalid date components or an
 *     unsupported format, all properties will be set to -1.
 */
```